### ISSUE TITLE
Improve a11y specifically to reach AA/AAA colour contrast for labeled buttons

### ISSUE THREAD

**Issue by luuu-xu** [activist-org/activist#215]
### Terms

- [X] I have searched [open and closed design issues](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Adesign)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

For our labeled buttons that is in brand orange colour, having white text inside does not meet WCAG colour contrast standards, see below:

<img width="327" alt="Screenshot 2023-07-06 at 21 57 35" src="https://github.com/activist-org/activist/assets/97932191/432aa843-1088-4b33-854a-26f470b4c122">

Changing the brand colour could be one solution, or we can have black text inside. 

And for our link within text, the colour blue is on the light side, we can make it a bit darker to reach AA rating, see below:

<img width="327" alt="Screenshot 2023-07-06 at 21 58 37" src="https://github.com/activist-org/activist/assets/97932191/5ebf3d23-5712-422b-92b8-dadf47a371fd">

### Contribution

I can make these easy changes once the design is considered and reached a conclusion.

**Comment by andrewtavis** [comment]
Reposting the tools that I generally use for this, @luuu-xu :)

- https://coolors.co/ for palettes and getting darker/lighter shades of colors
- https://polypane.app/color-contrast for a quick contrast check

**Comment by andrewtavis** [comment]
With each of these color changes we need to consider multiple backgrounds, specifically `content`, `distinct` (for cards and the sidebar content) and `header` (web and sidebar header/footer). Let's start with the link colors as they're maybe a bit more simple :) The alpha is definitely the part that's causing problems as it looks like when I checked I didn't add that in.

Based on checking some colors we could change `light-link-text` to `005CB8` while maintaining the alpha of 0.9, which would then give us AA accessibility over each of the background? 

If this sounds good you'd be welcome to open a PR. I've updated the [designs](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=805%3A231&mode=design&t=ApCWorDmDSuUVMt6-1) to use the new color and it's looking good :)

**Comment by andrewtavis** [comment]
By the looks of it our dark mode link text’s blue is already AA, so this PR would just be to fix the light mode color :)

**Comment by luuu-xu** [comment]
@andrewtavis Thank you for your ideas!

I agree with starting with `link-text` first. 

Shall we reach for AAA rating? If we'd like AAA, then I propose that we use these with 

1. light mode: `"link-text"`: `rgba(0, 72, 145, 0.9)` gives us **7.01**
2. light mode:  `"link-text-hover"`: `rgba(0, 47, 95, 0.9)` gives us **10.03**

If we are staying with AA, then your choice is great!

1. light mode: `"link-text"`: `rgba(0, 92, 184, 0.9)` gives us **5.31**
2. but we might wanna darker the hover color to `"link-text"`: `rgba(0, 59, 119, 0.9)` gives us **8.45**

I am also having a hard time changing the button colour because I like our orange theme colour, meanwhile the only way for a orange button to work with contrast is to have black text inside which is what we are doing with dark mode. 

Or we can change our orange colour to a darker, maybe brick colour, like `rgb(168, 96, 35)` but honestly I prefer the orange we have now.

**Comment by andrewtavis** [comment]
Really cool that it displays the colors! Didn't realize this was a feature 😊

> Shall we reach for AAA rating?

Maybe for this one we go for at least AA everywhere and then we eventually implement high contrast that gets it to AAA everywhere? I just checked `rgba(0, 72, 145, 0.9)` and over light distinct it's only 6.75 while not really looking particularly blue...

> 2. but we might wanna darker the hover color to `"link-text"`: `rgba(0, 59, 119, 0.9)` gives us 8.45

If you got this from the coolors shade picker this'd be fine 😊

> I am also having a hard time changing the button colour because I like our orange theme colour, meanwhile the only way for a orange button to work with contrast is to have black text inside which is what we are doing with dark mode.

This is exactly why I thought let's just do the link text for now :) Let's give it a thought in the mean time and we can discuss after the PR 🙃 

> Or we can change our orange colour to a darker, maybe brick colour, like `rgb(168, 96, 35)` but honestly I prefer the orange we have now.

Ya this is the problem with orange... I agree that the current version is better than the brick. Something will come out of this though!

Want to send along the link color PR then?

**Comment by andrewtavis** [comment]
Hey @luuu-xu 👋 Just a quick check with you on community matters :) First and foremost, we have the activist dev sync today at [15:00 UTC](https://zonestamp.toolforge.org/1689433242) at the following link: https://call.element.io/activist-dev-sync. Would be great to have you in there!

Also, I see that you're in the Matrix space, but you don't appear to be in any of the rooms. My suggestion would be to join the [Development](https://matrix.to/#/!CRgLpGeOBNwxYCtqmK:matrix.org?via=matrix.org&via=acter.global&via=chat.0x7cd.xyz) and [General](https://matrix.to/#/!uIGQUxlCnEzrPiRsRw:matrix.org?via=matrix.org&via=effektio.org&via=acter.global) rooms 🙃 Happy to have you bring some of the accessibility discussions in there so we can check with others and get more people thinking about this!

**Comment by andrewtavis** [comment]
Just sent you invites to both rooms 😊

**Comment by luuu-xu** [comment]
@andrewtavis Thank you for your invitation! I am sorry I couldn't make it to the dev sync I was busy during weekends. I just joined our Matrix space and read about the dev sync, I will continue communicating with you guys there. 😉

**Comment by andrewtavis** [comment]
Wonderful, @luuu-xu! So happy to have you there :)

**Comment by andrewtavis** [comment]
Continuing our main accessibility discussion, @luuu-xu :)

So I think it's now a forgone conclusion that we need to change the way that the buttons are colored in a pretty fundamental way. More specifically, orange cannot be the text on any button (or icon for that matter), as there really is no way to get it to be accessible without jumping hard towards an unappealing brick color. 

What it is I've found:

- `#F2A654` is roughly one shade lighter than our current `light-cta-orange` on [coolors](https://coolors.co/) ([link to palette with BG colors, F2A654 and the current light-cta-orange](https://coolors.co/ffffff-f0f0eb-f6f8fa-f2a654-f1993d))
  - This could be used as a background for buttons that have the `light-text` (`#000000` with an alpha of 0.85) as the text and the border color
  - This would get us to AAA! ([link to polypane comparison](https://polypane.app/color-contrast/#fg=rgba(0%2C%200%2C%200%2C%200.85)&bg=%23F2A654&level=aaa&format=hex&algo=WCAG2&filter=none))
  - How this would shift on hover is a question (likely just a bit lighter as we have it now?)
- The issue is that if we switch to using the text color, then now dark mode is problematic with white on orange 🤦‍♂️ 
- With this being said, for dark mode we're able to use the orange as a text color over the background 💡 
- What I'd suggest and would like your feedback on is the following system I've been thinking up:
  - We maintain the concept of certain special colors that we have
    - `cta-orange`, `act-red`, `learn-blue`, `accepted-green` (org has been accepted onto activist - we show a checkmark on org pages next to the name with this color and link to their application to see who vouched for them), `pending-yellow` (org is still applying and is thus filtered out from search results - org members can send the link to their application to others already on activist to be vouched for)
    - For elements associated with these colors:
      - In light mode the text color and border are the default text color (`#000000` with an alpha of 0.85), and the background is the special color
      - For dark mode the text color and border are the special color, and the background is the special color with a low alpha component applied to it

I've updated the first desktop image of light mode and dark mode on the [Home Pages tab on Figma](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=2475%3A19102&mode=design&t=6zv76m57Hbr15S88-1) and will post images of them below. Do you want to give them a look and let me know what you think? Definitely better than the ones we have now 😊 Your suggestions on hover actions would also be much appreciated!

### Light mode

![Topics Home - Light](https://github.com/activist-org/activist/assets/24387426/23d699c9-fe8f-4deb-92d5-c4329395f502)

### Dark mode

![Topics Home - Dark](https://github.com/activist-org/activist/assets/24387426/7ed821c5-49fd-4c6d-be97-9b14fa450f5c)

Once we have a final decision, I'll go ahead and edit the components on Figma as a final test and will then update the [tailwind.config.ts](https://github.com/activist-org/activist/blob/main/frontend/tailwind.config.ts) colors if everything is looking ok :) :)

**Comment by Vince155** [comment]
@andrewtavis I really like the look of dark mode. It's great that it achieves AA colour contrast too!

**Comment by andrewtavis** [comment]
> it is actually showing a range form 2.03 - 9.22 for when the background is cta-orange with 0.1 alpha, a warning says "Semi-transparent background! Showing range."

The colors themselves are not final, but I’d say we should make sure to check the color that’s generated by the aloha component over the background, which was coming out with daily good scores without me putting too much effort into it all. 

> I also bumped into articles taking about avoiding using semi-transparent backgrounds so I am not sure how we should proceed.

We could also not do the background at all for this, but I feel that even a 10-20% will make it look a bit more cohesive and not so 8-bit. 

In reference to the article, I definitely agree that we can maybe not worry as much when it comes to orange, but then I do generally find the current contrast doesn’t even pass the squint test for light mode… It does look nice, but then if we’re going to change it then I feel like the standards described get is to an easier situation for all the colors discussed, not just orange. 

Let me know if you’d change anything though, @luuu-xu! And thanks for checking this with us, @Vince155!! 😊😊

**Comment by luuu-xu** [comment]
Maybe the colour contrast standard does not apply or work well with semi-transparent background? See below where the dark mode button has only 1.89 ratings with `#F2A654` as text colour, and `#F2A654` with alpha of 0.1 as background. But to our eyes, it looks almost as contrast as with a black background which would have a rating of 9.87.

<img width="333" alt="Screenshot 2023-07-26 at 12 02 31" src="https://github.com/activist-org/activist/assets/97932191/b0c6c65d-c6cb-4b91-b568-2d3fee9e6925">

**Comment by andrewtavis** [comment]
Yes exactly, @luuu-xu :) When I check the accessibility of a transparent background I always look into what the actual color that’s represented on the page is when it’s applied over the background. For us that means seeing how it works when combined with content, distinct and header BG colors 😊

**Comment by andrewtavis** [comment]
I’ll check these changes with a few more folks and report back if we have the green light to implement from all stakeholders 🟢 Let me know if there are further reservations though!

**Comment by andrewtavis** [comment]
@luuu-xu, I've gone through and done an initial pass to update the colors of the [Figma designs](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=766%3A214&mode=design&t=r4gv42S7ZjmYTcUA-1). Would be great if you could provide some feedback 😊 Let's again remember that alpha backgrounds should be considered as their composite colors :)

Will also post this to the development channel. Will do the update in the code next week after we get the go ahead so we have it ready for Code Night 🌙🚀

**Comment by luuu-xu** [comment]
@andrewtavis I like the new design and buttons! I believe using black text inside orange button for light mode will be much better for contrast. And the dark mode buttons are passing AAA rating too. Great! 

Correct me if I am wrong, in dark mode, the new buttons do not have a hover effect right now? It seems like [figma](https://www.figma.com/file/I9McFfaLu1RiiWp5IP3YjE/activist_public_designs?type=design&node-id=1631-694&mode=design&t=I93kylF6kSGShp1h-0) has dark mode cta button and its hover state using the same colours. Maybe the hover state can have a higher alpha like `rgba(242, 166, 84, 0.2)` to highlight the hover state?

**Comment by andrewtavis** [comment]
Thanks for the note here, @luuu-xu! The hover action for dark mode was following a similar pattern to dark mode that ultimately doesn't make sense :) I've updated them to now have a 25% alpha of the Dark CTA Orange color on hover 😊

Will get to these updates after I merge the current PRs! 🚀

**Comment by andrewtavis** [comment]
f7fb36a closes this, @luuu-xu 😊 Was excited about checking it out, so I jumped into this before the PRs :) :) Thanks for the suggestions and all the work here! Not only are we more accessible, but the platform also looks much better now 🥳
